### PR TITLE
Issue/9796 filter product status

### LINKED ISSUE #9796: No Option to Filter Products
**Describe the bug**
The Products tab does not include an option to Filter products. I believe this may be a regression. The feature is available on iOS, but not on Android.

**To Reproduce**
1. Navigate to Products tab
2. There is no option to Filter products, only search.

**Expected behavior**
Merchants should have the option to Filter their products — by Stock Status, Product Status, Product Type, Product Category.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: Pixel 3
- Android version: 12
- WooCommerce Android version: 8.1.0

Above and below info is from my test setup, but it was **reported by a merchant** in 6795194-zd-a8c

**WordPress Environment**
<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://initialcharge.shop
Site address (URL): https://initialcharge.shop
WC Version: 8.1.0
REST API Version: ✔ 8.1.0
WC Blocks Version: ✔ 10.9.3
Action Scheduler Version: ✔ 3.6.2
Log Directory Writable: ✔
WP Version: 6.3.1
WP Multisite: –
WP Memory Limit: 768 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: Apache
PHP Version: 7.4.33
PHP Post Max Size: 256 MB
PHP Time Limit: 120
PHP Max Input Vars: 3000
cURL Version: 7.85.0
OpenSSL/3.0.9

SUHOSIN Installed: –
MySQL Version: 5.7.39-42-log
Max Upload Size: 256 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.1.0
WC Database Prefix: nyj_
Total Database Size: 7.54MB
Database Data Size: 5.32MB
Database Index Size: 2.22MB
nyj_woocommerce_sessions: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
nyj_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
nyj_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
nyj_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
nyj_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
nyj_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_actionscheduler_actions: Data: 0.06MB + Index: 0.13MB + Engine InnoDB
nyj_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_actionscheduler_logs: Data: 0.27MB + Index: 0.22MB + Engine InnoDB
nyj_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
nyj_jetpack_waf_blocklog: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_options: Data: 2.03MB + Index: 0.08MB + Engine InnoDB
nyj_postmeta: Data: 1.52MB + Index: 0.27MB + Engine InnoDB
nyj_posts: Data: 0.30MB + Index: 0.09MB + Engine InnoDB
nyj_sgs_log_events: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
nyj_sgs_log_visitors: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
nyj_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_usermeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
nyj_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
nyj_wc_admin_notes: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
nyj_wc_admin_note_actions: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
nyj_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
nyj_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
nyj_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
nyj_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
nyj_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
nyj_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
nyj_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
nyj_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
nyj_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
nyj_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 146
custom_css: 1
nav_menu_item: 15
page: 6
post: 1
product: 17
product_variation: 92
revision: 33
shop_order: 4

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (7) ###

Jetpack: by Automattic – 12.5
Printful Integration for WooCommerce: by Printful – 2.2.4
SiteGround Optimizer: by SiteGround – 7.4.1
Two Factor: by Plugin Contributors – 0.8.2
WooCommerce Payments: by Automattic – 6.4.2
WooCommerce Shipping & Tax: by WooCommerce – 2.3.4
WooCommerce: by Automattic – 8.1.0

### Inactive Plugins (0) ###


### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: donation (donation)
external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –
HPOS feature screen enabled: –
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: –

### WC Pages ###

Shop base: #21 - /
Cart: #22 - /cart/
Checkout: #23 - /checkout/
My account: #24 - /account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.5.2
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Subscriptions ###

WCS_DEBUG: ✔ No
Subscriptions Mode: ✔ Live
Subscriptions Live URL: https://initialcharge.shop
Subscriptions-core Library Version: 6.0.0
Subscription Statuses: –
WooCommerce Account Connected: ❌ No

### Store Setup ###

Country / State: United States (US) — New York

### Subscriptions by Payment Gateway ###


### Payment Gateway Support ###

WooPayments: products
refunds
multiple_subscriptions
subscription_cancellation
subscription_payment_method_change_admin
subscription_payment_method_change_customer
subscription_payment_method_change
subscription_reactivation
subscription_suspension
subscriptions
gateway_scheduled_payments
tokenization
add_payment_method

Cash on delivery: products

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: customize-store
minified-js
new-product-management-experience
product-variation-management
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2023-09-15 23:03:54 -05:00
Options: ✔
Notes: 126
Onboarding: completed

### WooCommerce Payments ###

Version: 6.4.2
Connected to WPCOM: Yes
Blog ID: 200282550
Account ID: acct_1K0Ysn2EdpuGw5Ex

### Action Scheduler ###

Complete: 71
Oldest: 2023-08-15 22:37:44 +0000
Newest: 2023-09-15 02:43:07 +0000

Failed: 9
Oldest: 2023-02-14 22:38:13 +0000
Newest: 2023-06-01 10:35:19 +0000

Pending: 1
Oldest: 2023-09-16 02:43:07 +0000
Newest: 2023-09-16 02:43:07 +0000


### Status report information ###

Generated at: 2023-09-15 13:17:30 -05:00
`
```
</details>

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #9796 

### Description

This PR fixes a bug in the product listing screen where the "Product Status" filter was not visible.

#### Reason
We were passing `NonPublishedProducts` product filter to the product filter screen. Applying that restriction basically says that we should only display published products on the product listing screen. When there's a restriction like this in the code for the product listing screen, it doesn't make sense to display a `product status` filter.

##### Fix
Remove the `NonPublishedProducts` product filter restriction on the product filtering screen.

### Testing instructions

Open the product listing screen, tap on the filter button on the top, and you must see the `product status` filter option.

### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@
 - [*] [Internal] In-Person Payments: Fixed wrong total payment amount display on the payment method selections screen when simple payments without tax used [https://github.com/woocommerce/woocommerce-android/pull/9826]
 - [*] In-Person Payments: Proactively notify users about optional card reader updates [https://github.com/woocommerce/woocommerce-android/pull/9847]
 - [*] Order creation: Customers list first page uses cached data [https://github.com/woocommerce/woocommerce-android/pull/9835]
+- [*] In-Person Payments: Fixed a product filter bug where users were not able to filter product by "Product Status" [https://github.com/woocommerce/woocommerce-android/pull/9891]
```
Comment (kidinov) [inline RELEASE-NOTES.txt]: np: is this IPP related fix?

**Review by kidinov — APPROVED** [review]
LGTM!

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (AnirudhBhat) [reply RELEASE-NOTES.txt]: Good catch! Fixed it here https://github.com/woocommerce/woocommerce-android/pull/9891/commits/9307aee49470a1f11d5f971b3e1c7cd5eb115e00
